### PR TITLE
fix(upload): #DRIV-66 import file with specials chars is now working

### PR DESCRIPTION
**Pull request by aymericdevriesere**
## Describe your changes
Upload files with special char is now working
## Checklist tests
- [x] Upload file with specials char in the nextcloud
- [x] Move the uploaded files under folder
- [x] Move uploaded files to workspace
- [x] Move uploaded files to root folder of nextcloud  
## Issue ticket number and link
[ DRIV-66 ]
https://jira.support-ent.fr/browse/DRIV-66
## Checklist before requesting a review (magic string, indentation, comment/documentation...)

- [x] I have detailed the tests to do in my feature/fix in order to prevent consequents regressions (must specify in **Checklist tests**)
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (API Doc etc...) - (must specify in **Description** for target version)
- [ ] If it is a consequent feature, I have added thorough tests.
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been added to this project (must specify in **Description**)

### CODE REVIEW

File: src/main/resources/public/ts/sniplets/content/workspace-nextcloud-content.sniplet.ts [hunk 1]
```diff
@@ -216,7 +216,7 @@ class ViewModel implements IViewModel {
         const selectedSet: Set<SyncDocument> = new Set(this.selectedDocuments);
         selectedSet.forEach((doc: SyncDocument) => {
             if (doc.path != target.path) {
-                promises.push(this.nextcloudService.moveDocument(model.me.userId, doc.path, (target.path != null ? target.path : "") + doc.name));
+                promises.push(this.nextcloudService.moveDocument(model.me.userId, doc.path, (target.path != null ? target.path : "") + encodeURI(doc.name)));
```
Comment (BastienBenard) [inline src/main/resources/public/ts/sniplets/content/workspace-nextcloud-content.sniplet.ts:219]: Ce serait plus de la responsabilité de ton service d'encoder l'url

File: src/main/resources/public/ts/sniplets/folder/nextcloud-folder.sniplet.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ export class FolderCreationModel implements IViewModel {
 
     public createFolder(folderCreate: models.Element): void {
         const folder: SyncDocument = this.vm.selectedFolder;
-        this.vm.nextcloudService.createFolder(model.me.userId, (folder.path != null ? folder.path + "/" : "") + folderCreate.name)
+        this.vm.nextcloudService.createFolder(model.me.userId, (folder.path != null ? folder.path + "/" : "") + encodeURI(folderCreate.name))
```
Comment (BastienBenard) [inline src/main/resources/public/ts/sniplets/folder/nextcloud-folder.sniplet.ts:44]: De meme

File: src/main/resources/public/ts/sniplets/content/workspace-nextcloud-content.sniplet.ts (hunk 1 above)
Comment (JinDouang) [reply src/main/resources/public/ts/sniplets/content/workspace-nextcloud-content.sniplet.ts:219]: A sa décharge, c'est un cas particulier

Le risque est que la méthode moveDocument ait un comportement différent sur d'autres sniplets qui appelles le service en ne souhaitant pas appeler avec le string encodeURI

On pourrait cependant envisager de dire que le service gère l'encodeURI mais qu'il soit géré dans un sniplet specifique (le content) ou service c'est ok car mapper le path avec et sans accent pour l'affichage de l'ent et l'appelle API via webdav Nextcloud c'était peut-être mal pensé...

File: src/main/resources/public/ts/sniplets/folder/nextcloud-folder.sniplet.ts (hunk 2 above)
Comment (JinDouang) [reply src/main/resources/public/ts/sniplets/folder/nextcloud-folder.sniplet.ts:44]: Même résonnement que dans mon commentaire en haut

**Review by JinDouang — APPROVED** [review]

**Pull request merged**